src/main/java/com/redhat/pantheon/servlet/AbstractJsonSingleQueryServlet.java

```diff
@@ -0,0 +1,69 @@
+package com.redhat.pantheon.servlet;
+
+import com.redhat.pantheon.jcr.JcrQueryHelper;
+import com.redhat.pantheon.model.QuerySingleResult;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nonnull;
+import javax.jcr.RepositoryException;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.Maps.newHashMap;
+import static com.redhat.pantheon.servlet.ServletUtils.writeAsJson;
+
+/**
+ * A base servlet that helps to create simple Json result from custom queries.
+ * This servlet should be extended to create components that render a single element of
+ * results in json format. See methods to be overriden for details on how to
+ * customize sub classes.
```

Comment:
Comments probably need to be updated as they seem like a copy-paste from the multi-result query servlet. Since you are executing any custom query, I would include something about how the servlet behaves when it finds more than one result. i.e. it renders the first one.

Reply:
Good point.

Reply:
changed